[pytorch] update CI script to take string docker image version

ljk53 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #30937 [pytorch] integrate op dependency analysis process into CI
* #31848 [pytorch] update docker image version
* **#31857 [pytorch] update CI script to take string docker image version**

Summary:
According to @mingbowan we will change to use string docker image
version because the tag is no longer an integer since we move the docker
image build job to circle CI:
http://ossci-docker.s3-website.us-east-1.amazonaws.com/pytorch.html

Test Plan:
- with stacked PR

Differential Revision: [D19282726](https://our.internmc.facebook.com/intern/diff/D19282726)

ljk53 committed:
[pytorch] update CI script to take string docker image version

Summary:
According to @mingbowan we will change to use string docker image
version because the tag is no longer an integer since we move the docker
image build job to circle CI:
http://ossci-docker.s3-website.us-east-1.amazonaws.com/pytorch.html

Test Plan:
- with stacked PR

[ghstack-poisoned]

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit 04203db1:


**None of the build failures appear to be your fault**.

* **1/1** recognized as flaky :snowflake:
    * Re-run these jobs?


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=04203db1233342a38fc46bf2c2aaf3a3c299bc6e).

### :snowflake: 1 failure recognized as flaky
The following build failures have been detected as flaky and **may not be your fault**:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4132882) pytorch_linux_xenial_cuda9_cudnn7_py3_test (1/1)
**Step:** "Test" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=80373386) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=80373386)) :snowflake:
<details>
<summary>
<code>Jan 04 18:41:13 AssertionError: 1024 not less than or equal to 1e-05 : __main__.TestAutogradDeviceTypeCUDA.test_logdet_1x1_cuda leaked 1024 bytes CUDA memory on device 0</code>
</summary>

```
Jan 04 18:41:13 ====================================================================== 
Jan 04 18:41:13 FAIL [0.108s]: test_logdet_1x1_cuda (__main__.TestAutogradDeviceTypeCUDA) 
Jan 04 18:41:13 ---------------------------------------------------------------------- 
Jan 04 18:41:13 Traceback (most recent call last): 
Jan 04 18:41:13   File "/var/lib/jenkins/workspace/test/common_utils.py", line 665, in wrapper 
Jan 04 18:41:13     method(*args, **kwargs) 
Jan 04 18:41:13   File "/var/lib/jenkins/workspace/test/common_utils.py", line 521, in __exit__ 
Jan 04 18:41:13     self.name, after - before, i)) 
Jan 04 18:41:13   File "/var/lib/jenkins/workspace/test/common_utils.py", line 877, in assertEqual 
Jan 04 18:41:13     super(TestCase, self).assertLessEqual(abs(x - y), prec, message) 
Jan 04 18:41:13 AssertionError: 1024 not less than or equal to 1e-05 : __main__.TestAutogradDeviceTypeCUDA.test_logdet_1x1_cuda leaked 1024 bytes CUDA memory on device 0 
Jan 04 18:41:13  
Jan 04 18:41:13 ---------------------------------------------------------------------- 
Jan 04 18:41:13 Ran 1884 tests in 880.920s 
Jan 04 18:41:13  
Jan 04 18:41:13 FAILED (failures=1, skipped=13, expected failures=1) 
Jan 04 18:41:13  
Jan 04 18:41:13 Generating XML reports... 
Jan 04 18:41:14 Traceback (most recent call last): 
Jan 04 18:41:14   File "test/run_test.py", line 456, in <module> 
Jan 04 18:41:14     main() 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

ljk53 committed:
Update on "[pytorch] update CI script to take string docker image version"


Summary:
According to @mingbowan we will change to use string docker image
version because the tag is no longer an integer since we move the docker
image build job to circle CI:
http://ossci-docker.s3-website.us-east-1.amazonaws.com/pytorch.html

Test Plan:
- with stacked PR

Differential Revision: [D19282726](https://our.internmc.facebook.com/intern/diff/D19282726)

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] update CI script to take string docker image version"


Summary:
According to @mingbowan we will change to use string docker image
version because the tag is no longer an integer since we move the docker
image build job to circle CI:
http://ossci-docker.s3-website.us-east-1.amazonaws.com/pytorch.html

Test Plan:
- with stacked PR

Differential Revision: [D19282726](https://our.internmc.facebook.com/intern/diff/D19282726)

[ghstack-poisoned]
